ArduCopter/takeoff.cpp

```diff
@@ -131,6 +133,12 @@ void Mode::auto_takeoff_run()
 
     // aircraft stays in landed state until rotor speed run up has finished
     if (motors->get_spool_state() == AP_Motors::SpoolState::THROTTLE_UNLIMITED) {
+        // Ensure position controller has been initialized.  This catches Heli can entering
+        // takeoff from a throttle unlimitied state and skipping the initialization.
+        if (!pos_control->is_active_xy()) {
```

Comment:
This should not be needed. We only call this from Auto and Guided so both should already be initialised.

How do we get here in throttle unlimited without initialising? That is where we need to fix this.